src/main/kotlin/com/dto/ReservationManager.kt

```diff
@@ -0,0 +1,61 @@
+package com.dto
+
+import com.excpetion.ReservationAlreadyStartedException
+import com.excpetion.ReservationNotFoundException
+import com.model.ReservationDao
+import com.parkingrate.ParkingRateEvaluator
+import com.util.ReservationCalculator
+import org.springframework.stereotype.Service
+import java.math.BigDecimal
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.util.Optional
+import javax.transaction.Transactional
+
+
+@Service
+class ReservationManager(private val reservationDao: ReservationDao,
+                         private val reservationConverter: ReservationConverter,
+                         private val reservationCalculator: ReservationCalculator) {
+
+    @Transactional
+    fun createReservation(reservationRequest: ReservationRequest, startTime: LocalDateTime): StartedReservationDto {
+
+        Optional.ofNullable(reservationDao.findStartedReservation(reservationRequest.carLicensePlate))
+                .ifPresent { throw ReservationAlreadyStartedException(reservationRequest.carLicensePlate) }
+
+        return reservationConverter.convertToModel(reservationRequest, startTime)
+                .also { reservationDao.save(it) }
+                .let { reservationConverter.convertToReservationStartedDto(it) }
+    }
+
+    @Transactional
+    fun stopReservation(reservationId: Long, endTime: LocalDateTime): StoppedReservationDto {
+        val reservation = reservationDao.findById(reservationId)
+
+        return reservation
+                .map {
+                    it.endTime = endTime
+                    it.cost = reservationCalculator.calculateReservationCost(
+                            parkingRate = ParkingRateEvaluator.retrieveParkingRatesForDriver(reservation.get().driverType),
```

Comment:
dlaczego `reservation.get()`, a nie `it`?

Reply:
racja powinienem się trzymać konwencji, zmiana bez powodu.